syntax = "proto3";

option java_package = "org.oliverlittle.clusterprocess";

message DataSource {
    oneof source {
        CassandraDataSource cassandra = 1;
    }
}

message CassandraDataSource {
    string server_url = 1;
    string keyspace = 2;
    string table = 3;
}

message CassandraTokenRange {
    sint64 start = 1;
    sint64 end = 2;
}

message InetSocketAddress {
    string host = 1;
    int32 port = 2;
}